### ISSUE TITLE
Web component/React: Leaving IBM - voiceover iPhone, the screen reader focus is not trapped within the dialog when we navigate using Left/Right swipe. 

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#5061]
## Detailed description

Issue:
With voiceover iPhone, the screen reader focus is not trapped within the dialog when we navigate using Left/Right swipe. 

Expected:
Screen reader focus should be trapped within the dialog while navigating using Left/Right swipe

**Comment by IgnacioBecerra** [comment]
@Praveen-sr Wondering if you can provide a video? I can't seem to replicate this behavior on my end. 

cc: @RobertaJHahn

**Comment by Praveen-sr** [comment]
@IgnacioBecerra As we discussed on the call yesterday, the dev thinks that this could be a story book issue.
He would update after discussing with the other engineers.
